{
  try {
    doRun();
  }
 catch (  RuntimeException e) {
    exceptionThrownByThread=e;
  }
}
